File: C/Algorithm_DataStructure/20_01/Stack_byList/stack.c
#include "stack.h"
#include <stdio.h>
#include <stdlib.h>

Stack MakeStack(void)
{
    return MakeList();
}

bool isEmpty(Stack stack)
{
    return IsEmpty(stack);
}

static Element topElement(const Stack stack)
{
    Element thisElement = stack;
    while (thisElement->nextPtr)
    {
        thisElement = thisElement->nextPtr;
    }
    return thisElement;
}

int top(Stack stack)
{
    Element top = topElement(stack);
    return top->num;
}

bool changeTopValues(Stack stack, int newValue)
{
    Element top = topElement(stack);
    int oldValue = top->num;
    top->num = newValue;
    printf("Top value changed from %d to %d.\n", oldValue, newValue);
    return 1;
}

bool push(Stack stack, int ToPush)
{
    Append(stack, ToPush);
    return 1;
}

bool pop(Stack stack)
{
    if (isEmpty(stack))
    {
        printf("The stack is already empty!\n");
        return 0;
    }

    Element prev = stack, this, temp;
    this = prev->nextPtr;

    while (this->nextPtr)
    {
        temp = this;
        this = prev->nextPtr;
        prev = temp;
    }

    prev->nextPtr = NULL;
    free(this);
    return 1;
}

int height(Stack stack)
{
    return len(stack);
}

File: Python/course_work/Code for the stock price problem (1)/my/stack.py
class Stack:
    def __init__(self):
        self.items = []

    def isEmpty(self):
        return self.items == []

    def hasItem(self):
        return self.items != []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def peek(self):
        """return the top element of a stack"""
        return self.items[len(self.items)-1]

    def size(self):
        """return the height of stack"""
        return len(self.items)
